pkg/resources/credentials.go

```diff
@@ -660,6 +660,16 @@ func GetVSphereCredentials(data CredentialsData) (VSphereCredentials, error) {
 			return VSphereCredentials{}, err
 		}
 	}
+	if username == "" {
```

Comment:
One thing that I am wondering about which might not be relevant for this PR is, the InfraManagement credentials, do they some how end up in the machine deployment, or being used in machine controller ?

Reply:
The credentials are exposed to the machine-controller via env-var read from `cloud-credentials` secret.